java/statestore-commit/src/main/java/sleeper/commit/StateStoreCommitter.java

```diff
@@ -120,4 +123,8 @@ private void apply(IngestAddFilesCommitRequest request) throws StateStoreExcepti
         }
     }
 
+    private void apply(CompactionJobIdAssignmentCommitRequest request) throws StateStoreException {
+        StateStore stateStore = stateStoreProvider.getByTableId(request.getTableId());
+        stateStore.assignJobIds(request.getAssignJobIdRequests());
```

Comment:
This can fail if the files have already been assigned to a different job. Currently that would cause the message to be put back on the queue and retried, and it would fail repeatedly. We can log the failure here and ignore it instead. Please can we add a test case for this?

Reply:
Thanks, added a test for this